ccm_web/client/src/pages/Home.tsx

```diff
@@ -18,48 +22,101 @@ const useStyles = makeStyles((theme) => ({
   },
   featureCardContainer: {
     padding: 5
+  },
+  courseName: {
+    textAlign: 'left',
+    padding: 5
   }
 }))
 
-function FeatureGroup (featureGroup: FeatureUIGroup): JSX.Element {
-  const classes = useStyles()
-  return (
-    <Grid key={featureGroup.id} container item xs={12} spacing={0}>
-      <Grid item xs={12}><Typography variant='h6' className={classes.title} >{featureGroup.title}</Typography></Grid>
-      <Grid container item xs={12}>
-        {featureGroup.features.sort((a, b) => (a.data.ordinality < b.data.ordinality) ? -1 : 1).map(feature => {
-          return (
-            <Grid key={feature.data.id} item className={classes.featureCardContainer} xs={12} sm={4}>
-              <FeatureCard {...feature} />
-            </Grid>
-          )
-        })}
-      </Grid>
-    </Grid>
-  )
+interface HomeProps extends LtiProps {
+  ltiKey: string | undefined
 }
 
-function Home (): JSX.Element {
+function Home (props: HomeProps): JSX.Element {
+  const [globals, isAuthenticated, loading, error] = useGlobals(props.ltiKey)
   const classes = useStyles()
-  const saveCourseName = async (courseName: string): Promise<void> => await new Promise<void>((resolve, reject) => {
-    console.log('saveCourseName ' + courseName)
-    if (courseName === 'reject') {
-      reject(new Error('Rejected as requested'))
+
+  const renderFeature = (feature: FeatureUIProps): JSX.Element => {
+    return (
+      <Grid key={feature.data.id} item className={classes.featureCardContainer} xs={12} sm={4}>
+        <FeatureCard {...feature} />
+      </Grid>
+    )
+  }
+
+  const renderFeatureGroup = (featureGroup: FeatureUIGroup): JSX.Element => {
+    return (
+      <Grid key={featureGroup.id} container item xs={12} spacing={0}>
+        <Grid item xs={12}><Typography variant='h6' className={classes.title} >{featureGroup.title}</Typography></Grid>
+        <Grid container item xs={12}>
+          {featureGroup.features.sort((a, b) => (a.data.ordinality < b.data.ordinality) ? -1 : 1)
+            .filter(feature => {
+              return isAuthorizedForFeature((globals != null) ? globals.course.roles : [], feature)
+            })
+            .map(feature => {
+              return renderFeature(feature)
+            })}
+        </Grid>
+      </Grid>
+    )
+  }
+
+  const renderLoading = (): JSX.Element => {
+    return (<span>Loading</span>)
+  }
+
+  const renderNotAuthenticated = (): JSX.Element => {
+    return (<span>Not Authenticated</span>)
+  }
+
+  const renderError = (): JSX.Element => {
+    return (<div id='error' hidden={error === undefined}>Error: {error?.message}</div>)
+  }
```

Comment:
I think if we take the approach I suggested above in letting `App` handle fetching globals, some or all of this may not be necessary. You'll be able to assume it's loaded and the user is authenticated if the `Home` component is being rendered.

Reply:
@ssciolla ok I passed globals in to the props for home and removed this

Reply:
Cool, I will re-review and run it again locally, thanks